runtime/compilers/rillv1/parse_api.go

```diff
@@ -88,20 +98,23 @@ func (p *Parser) parseAPI(node *Node) error {
 	r.APISpec.OpenapiSummary = openapiSummary
 	r.APISpec.OpenapiParameters = openapiParams
 	r.APISpec.OpenapiResponseSchema = openapiSchema
+	r.APISpec.SecurityRules = securityRules
 
 	return nil
 }
 
 // DataYAML is the raw YAML structure of a sub-property for defining a data resolver and properties.
 // It is used across multiple resources, usually under "data:", but inlined for APIs.
 type DataYAML struct {
-	Connector      string         `yaml:"connector"`
-	SQL            string         `yaml:"sql"`
-	MetricsSQL     string         `yaml:"metrics_sql"`
-	API            string         `yaml:"api"`
-	Args           map[string]any `yaml:"args"`
-	Glob           yaml.Node      `yaml:"glob"` // Path (string) or properties (map[string]any)
-	ResourceStatus map[string]any `yaml:"resource_status"`
+	Connector            string             `yaml:"connector"`
+	SQL                  string             `yaml:"sql"`
+	MetricsSQL           string             `yaml:"metrics_sql"`
+	API                  string             `yaml:"api"`
+	Args                 map[string]any     `yaml:"args"`
+	Glob                 yaml.Node          `yaml:"glob"` // Path (string) or properties (map[string]any)
+	ResourceStatus       map[string]any     `yaml:"resource_status"`
+	Security             SecurityPolicyYAML `yaml:"security"`
+	SkipResolverSecurity bool               `yaml:"skip_resolver_security"`
```

Comment:
I think these should be defined on the `APIYAML` type since they are specific to it – the `DataYAML` type is a partial which is used by multiple resources (and should probably be moved to its own file, similar to `parse_partial_field_selector.go`, though you don't need to do that now)

Reply:
Ahh, agreed. Moving